graph/temporal.go

```diff
@@ -0,0 +1,17 @@
+package graph
+
+type TemporalLine interface {
+	Line
```

Comment:
The `Line.ID` method is never used here, so this could actually be `Edge`.

Reply:
But given that temporal extends directed multi, it should satisfy the `Line` interface for consistency, no?